src/internal/m365/exchange/events_restore.go

```diff
@@ -210,15 +212,108 @@ func updateExceptionOccurrences(
 
 		evt = toEventSimplified(evt)
 
-		// TODO(meain): Update attachments (might have to diff the
-		// attachments using ids and delete or add). We will have
-		// to get the id of the existing attachments, diff them
-		// with what we need a then create/delete items kinda like
-		// permissions
 		_, err = ac.PatchItem(ctx, userID, ptr.Val(evts[0].GetId()), evt)
 		if err != nil {
 			return clues.Wrap(err, "updating event instance")
 		}
+
+		evt, err = api.EventFromMap(instance)
+		if err != nil {
+			return clues.Wrap(err, "parsing exception event (second)")
+		}
+
+		err = updateAttachments(ctx, ac, userID, containerID, ptr.Val(evts[0].GetId()), evt)
+		if err != nil {
+			return clues.Wrap(err, "updating attachments")
+		}
+	}
+
+	return nil
+}
+
+func updateAttachments(
+	ctx context.Context,
+	client api.Events,
+	userID, containerID, eventID string,
+	event models.Eventable,
+) error {
+	// Attachment can only be deleted or added in modified events and
+	// when an event is modified, the attachment ids will be
+	// changed. This change is both present in the data that we backed
+	// up and so we cannot rely on the attachment id to decide if they
+	// are the same. Instead we make use of the contentBytes and name
+	// to ensure they are the same.
+	attachments, err := client.GetAttachments(ctx, false, userID, eventID)
+	if err != nil {
+		return clues.Wrap(err, "getting attachments")
+	}
+
+	// Delete unnecessary attachments
```

Comment:
This comment could use more explanation to describe the overall flow and purpose of the loop.  It may be a small and relatively straightforward algorithm, but since it's juggling two different sources with effectively the same data, it can be easy to mix up the state of the two when constructing a mental state flow.